lib/hirefire/macro/que.rb

```diff
@@ -20,17 +16,40 @@ module Que
       # @return [Integer] the number of jobs in the queue(s).
       #
       def queue(*queues)
-        query = case
-        when queues.none? then QUERY
-        when queues.one? then "#{QUERY} AND queue = '#{queues.first}'"
-        else
-          queue_names = queues.map { |queue| "'#{queue}'" }.join(', ')
-          %Q{#{QUERY} AND queue IN (#{queue_names})}
-        end
-
+        query   = queues.empty? && base_query || base_query + " AND queue IN (#{names(queues)})"
         results = ::Que.execute(query).first
         (results[:total] || results["total"]).to_i
       end
+
+      private
+
+      def base_query
+        return QUE_V0_QUERY if defined?(::Que::Version)
+        return QUE_V1_QUERY if defined?(::Que::VERSION)
+        raise "Couldn't find Que version"
+      end
+
+      def names(queues)
+        queues.map { |queue| "'#{queue}'" }.join(",")
+      end
+
+      def query_const(query)
+        query.gsub(/\s+/, " ").strip.freeze
+      end
+
+      QUE_V0_QUERY = query_const(<<-QUERY)
+        SELECT count(*) AS total
```

Comment:
use same case as query below

```suggestion
        SELECT COUNT(*) AS total
```